src/python_inspector/resolution.py

```diff
@@ -743,6 +743,7 @@ def get_resolved_dependencies(
     verbose: bool = False,
     pdt_output: bool = False,
     analyze_setup_py_insecurely: bool = False,
+    ctx=None,
```

Comment:
Is this really needed? I get why you do this, but IMHO it would be simpler to pass down a "printer" as explained below and more explicit flags if needed